rfcs/0000-consistent-resource-names.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: project-name-rationalization
+- Start Date: 2022-05-16
+- RFC PR: [mbta/technology-docs#13](https://github.com/mbta/technology-docs/pull/13)
+- Asana task: [RFC: rationalizing how we name projects](https://app.asana.com/0/1113179098808463/1202191100361265/f)
+- Status: Proposed
+
+# Summary
+[summary]: #summary
+
+Resource names in CTD have a consistent pattern.
+
+# Motivation
+[motivation]: #motivation
+
+As CTD has grown, multiple teams have independently named their projects and the associated resources. As these names have been generated independently, they are not always consistent about how they use punctuation to separate multiple words in a resource name, leading to confusion.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Within CTD, we use one of the following patterns for naming projects and resources:
+- kebab-case: all lowercase, separated by hyphens
+- snake_case: all lowercase, separated by underscores
+- camelCase: first word is lower case, other words are Title case with no separation
+- PascalCase: all words are Title case with no separation
+
+We use the following patterns for resources:
```

Comment:
What is "current practice" will change as the recommendations in this RFC are adopted, so leaving that out allows the document to avoid being out-of-date in the future.